plugin-locationlayer/src/main/java/com/mapbox/mapboxsdk/plugins/locationlayer/LocationLayerPlugin.java

```diff
@@ -513,8 +513,12 @@ private void initialize() {
     mapboxMap.addOnMapLongClickListener(onMapLongClickListener);
 
     locationLayer = new LocationLayer(mapView, mapboxMap, options);
-    locationLayerCamera = new LocationLayerCamera(
-      mapView.getContext(), mapboxMap, cameraTrackingChangedListener, options, onCameraMoveInvalidateListener);
+
+    CameraUpdateProvider cameraUpdateProvider = new CameraUpdateProvider(mapboxMap,
```

Comment:
This provider's queue needs to be cleared similarly to `StaleStateManager#onStop`.